test/index.js

```diff
@@ -1463,6 +1463,139 @@ describe('Nightmare', function () {
       didSubscribe.should.be.false;
     });
 
+    it('should subscribe to frames when requested necessary', function(done) {
+      var didSubscribe = false;
+      var didUnsubscribe = false;
+      var FrameManager = require('../lib/frame-manager.js');
+      var fn;
+      var manager = FrameManager({
+        webContents: {
+          debugger: {
+            isAttached: function() { return true; },
+            sendCommand: function(command) { if (command === 'DOM.highlightRect') { fn('mock-data'); }}
+          },
+          beginFrameSubscription: function(_fn) { didSubscribe = true; fn = _fn; },
+          endFrameSubscription: function() { didUnsubscribe = true; },
+          executeJavaScript: function() {}
+        }
+      });
```

Comment:
I'm on the fence about this. I like the idea but...

- We define custom hooks in each of the tests so it won't save us too many lines of code. It might even lead to more confusion due to jumping from code to definition and back. That is:

```
var manager = FrameManager({
  webContents: {
    debugger: {
      isAttached: function() { return true; },
      sendCommand: function(command) { if (command === 'DOM.highlightRect') { fn('mock-data'); }}
    },
    beginFrameSubscription: function(_fn) { didSubscribe = true; fn = _fn; },
    endFrameSubscription: function() { didUnsubscribe = true; },
    executeJavaScript: function() {}
  }
});
```

will become:

```
var mockWindow = MockWindow({
  // Unsure about how we define `fn` access at the moment
  sendCommand: function(command) { if (command === 'DOM.highlightRect') { this.fn('mock-data'); }}
  beginFrameSubscription: function() { didSubscribe = true; },
  endFrameSubscription: function() { didUnsubscribe = true; }
})
```

- This file is quite lengthy and isn't broken down, where would the class definition live? My gut says top of the file or in a separate file but both feel a little premature/less clear when reading code